In [9]:
import pandas as pd
import numpy as np
from scipy import stats

## Closed_deals

In [10]:
# Dataset load and samples
Closed_deals = pd.read_csv('../Datasets/Datasets_original/olist_closed_deals_dataset.csv')
Closed_deals.sample(10)

mql_id                         seller_id  \
380  3da19335c0dfd2c484d977ca16bfebd2  5bf397d04035b5f45cc41e82ec106803   
714  90da5fb6873f5daa02586c51fec88189  9f15313cc96cd611d45e63eab4170bf1   
826  98826328c3a60215e53b342560084571  f46490624488d3ff7ce78613913a7711   
114  b30be338efa14f74a40949dbafd02b57  c8cc60ffd415254af5d9b9f34f32c369   
292  41fb0fd43593de7d24448de9f5aea38e  b25ec1aff84400104d63eadc3b3052fa   
805  2852475bf1d02022e4eb5269c452259a  1a47a5831effd926ad6bee7ab8beb86c   
511  dada30935bcfc75c69e651430ea8815f  f2c87addc7e39e1c73457c4ec85ed728   
727  2c30dbced2ca2da386510d7416679269  d598f929fc44e1e38678e7f47250ec04   
28   9ce2e101957dc53e3e89cc8e1599aa60  a4fc6eadaff1576af888509a53420f6a   
105  42e452a93e736373dd9281c113eff2b7  da6a60cc8cc724fe51be021ff8be779c   

                               sdr_id                             sr_id  \
380  068066e24f0c643eb1d089c7dd20cd73  2695de1affa7750089c0455f8ce27021   
714  4b339f9567d060bcea4f5136b9f5949e  060c0a26f19f4d66b42e0d8796688490   
826  9d12ef1a7eca3ec58c545c678af7869c  068066e24f0c643eb1d089c7dd20cd73   
114  068066e24f0c643eb1d089c7dd20cd73  fbf4aef3f6915dc0c3c97d6812522f6a   
292  9e4d1098a3b0f5da39b0bc48f9876645  2695de1affa7750089c0455f8ce27021   
805  370c9f455f93a9a96cbe9bea48e70033  495d4e95a8cf8bbf8b432b612a2aa328   
511  33dd30594303eb0ebe12a556d612c437  4b339f9567d060bcea4f5136b9f5949e   
727  56bf83c4bb35763a51c2baab501b4c67  495d4e95a8cf8bbf8b432b612a2aa328   
28   09285259593c61296eef10c734121d5b  6565aa9ce3178a5caf6171827af3a9ba   
105  0a0fb2b07d841f84fb6714e35c723075  495d4e95a8cf8bbf8b432b612a2aa328   

                won_date         business_segment        lead_type  \
380  2018-03-13 17:16:23      household_utilities          offline   
714  2018-03-02 12:58:05          car_accessories    online_medium   
826  2018-05-16 19:51:05            health_beauty       online_big   
114  2018-04-30 20:23:10                      pet    online_medium   
292  2018-03-19 20:04:46  audio_video_electronics          offline   
805  2018-06-29 17:54:16               home_decor    online_medium   
511  2018-09-27 17:36:04           bed_bath_table    online_medium   
727  2018-04-20 18:13:06          car_accessories         industry   
28   2018-04-13 12:27:48                computers  online_beginner   
105  2018-05-14 16:06:59                    books       online_big   

    lead_behaviour_profile has_company has_gtin average_stock business_type  \
380                    cat         NaN      NaN           NaN      reseller   
714                   wolf         NaN      NaN           NaN      reseller   
826                  eagle         NaN      NaN           NaN      reseller   
114                    cat         NaN      NaN           NaN      reseller   
292                    cat         NaN      NaN           NaN      reseller   
805                   wolf         NaN      NaN           NaN      reseller   
511                    cat       False     True        50-200      reseller   
727                    cat         NaN      NaN           NaN      reseller   
28                     cat         NaN      NaN           NaN      reseller   
105                    cat         NaN      NaN           NaN      reseller   

     declared_product_catalog_size  declared_monthly_revenue  
380                            NaN                       0.0  
714                            NaN                       0.0  
826                            NaN                       0.0  
114                            NaN                       0.0  
292                            NaN                       0.0  
805                            NaN                       0.0  
511                           85.0                       0.0  
727                            NaN                       0.0  
28                             NaN                       0.0  
105                            NaN                       0.0

In [11]:
Closed_deals.shape

(842, 14)

In [12]:
Closed_deals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    object 
 1   seller_id                      842 non-null    object 
 2   sdr_id                         842 non-null    object 
 3   sr_id                          842 non-null    object 
 4   won_date                       842 non-null    object 
 5   business_segment               841 non-null    object 
 6   lead_type                      836 non-null    object 
 7   lead_behaviour_profile         665 non-null    object 
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     object 
 11  business_type                  832 non-null    object 
 12  declared_product_catalog_size  69 non-null     flo

In [13]:
# Description
Closed_deals.describe()

declared_product_catalog_size  declared_monthly_revenue
count                      69.000000              8.420000e+02
mean                      233.028986              7.337768e+04
std                       352.380558              1.744799e+06
min                         1.000000              0.000000e+00
25%                        30.000000              0.000000e+00
50%                       100.000000              0.000000e+00
75%                       300.000000              0.000000e+00
max                      2000.000000              5.000000e+07

In [14]:
# Non-numeric description
Closed_deals.describe(include=['O'])

mql_id                         seller_id  \
count                                842                               842   
unique                               842                               842   
top     5420aad7fec3549a85876ba1c529bd84  2c43fb513632d29b3b58df74816f1b06   
freq                                   1                                 1   

                                  sdr_id                             sr_id  \
count                                842                               842   
unique                                32                                22   
top     4b339f9567d060bcea4f5136b9f5949e  4ef15afb4b2723d8f3d81e51ec7afefe   
freq                                 140                               133   

                   won_date business_segment      lead_type  \
count                   842              841            836   
unique                  824               33              8   
top     2018-05-04 03:00:00       home_decor  online_medium   
freq                      6              105            332   

       lead_behaviour_profile has_company has_gtin average_stock business_type  
count                     665          63       64            66           832  
unique                      9           2        2             6             3  
top                       cat        True     True          5-20      reseller  
freq                      407          58       54            22           587

- 842 rows. has_company, has_gtin, average_stock and  declared_product_catalog_size have more than 90 % of missing data
- This table is not very relevant

## Customers

In [15]:
# Dataset load and samples
Customers = pd.read_csv('../Datasets/Datasets_original/olist_customers_dataset.csv')
Customers.sample(10)

customer_id                customer_unique_id  \
33283  ad5e33166ef8e090f857ea53dc70bb18  80172250f4b01859c09d7308e3090a5f   
28791  740e4bdbbba1f7da2c2819d324fdc3cf  2c683405d8605cba6348bbca4d22bf6e   
49507  bfe964865bb435b81a560ba893502540  248acdcc29adc0f9e144da53c94569f3   
17335  1dbf434ac2c0271c0a5222be25c8fe62  a98127f5902b8118fd702de874f1287c   
69954  dd0ebe50a2f4bf520764807c1d5ab294  8a3e2983c35b5d2fb78686794986f7da   
85778  64614b6f07b27eea08eac0f2c527645a  b7d0720db8231cc62b761c091f9045f8   
68966  0b35cf387e2b991b1a6928a3cb351f0a  0b21a574354d2ed360d0b1b4762e41c9   
94138  98f2de76457ee90e5d8b763c25c734de  10276d60d5486cb3e368bfa68c5351d3   
81229  3d7697c0971cb2606444adae322d8bb3  f4513d9cd4f25a8254176ceda0abdc5e   
84951  ab85c1abaa5e143a652cb4f298fa5776  e0c3bef7c7e5679c7776c40abec7fe2b   

       customer_zip_code_prefix          customer_city customer_state  
33283                     29645  santa maria de jetiba             ES  
28791                     87502               umuarama             PR  
49507                     13290               louveira             SP  
17335                     79602            tres lagoas             MS  
69954                      3026              sao paulo             SP  
85778                     16902              andradina             SP  
68966                     12246    sao jose dos campos             SP  
94138                     79560        chapadao do sul             MS  
81229                     89835           sao domingos             SC  
84951                     12235    sao jose dos campos             SP

In [16]:
Customers.shape

(99441, 5)

In [17]:
Customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [18]:
# Description
Customers.describe()

customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

In [19]:
# Non-numeric description
Customers.describe(include=['O'])

customer_id                customer_unique_id  \
count                              99441                             99441   
unique                             99441                             96096   
top     06b8999e2fba1a1fbc88172c00ba8bc7  8d50f5eadf50201ccdcedfb9e2ac8455   
freq                                   1                                17   

       customer_city customer_state  
count          99441          99441  
unique          4119             27  
top        sao paulo             SP  
freq           15540          41746

- Missing data in customers_unique_id without reference
- This table is helpful to locate the customers

## Geolocation

In [20]:
# Dataset load and samples
Geolocation = pd.read_csv('../Datasets/Datasets_original/olist_geolocation_dataset.csv')
Geolocation.sample(10)

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
896681                        87310       -24.028721       -52.355347   
817508                        77405       -11.728625       -49.068673   
363305                        16480       -21.910409       -49.896353   
800114                        74603       -16.685721       -49.236043   
937418                        89980       -26.396149       -53.081461   
313835                        13503       -22.404861       -47.578636   
887455                        86390       -23.034025       -50.076683   
692513                        45608       -14.815430       -39.277651   
305341                        13360       -23.000887       -47.517216   
455250                        22793       -22.999492       -43.409301   

       geolocation_city geolocation_state  
896681     campo mourao                PR  
817508           gurupi                TO  
363305          guaimbê                SP  
800114          goiânia                GO  
937418        campo ere                SC  
313835        rio claro                SP  
887455          cambará                PR  
692513          itabuna                BA  
305341         capivari                SP  
455250   rio de janeiro                RJ

In [21]:
Geolocation.shape

(1000163, 5)

In [22]:
Geolocation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


In [23]:
# Description
Geolocation.describe()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng
count                 1.000163e+06     1.000163e+06     1.000163e+06
mean                  3.657417e+04    -2.117615e+01    -4.639054e+01
std                   3.054934e+04     5.715866e+00     4.269748e+00
min                   1.001000e+03    -3.660537e+01    -1.014668e+02
25%                   1.107500e+04    -2.360355e+01    -4.857317e+01
50%                   2.653000e+04    -2.291938e+01    -4.663788e+01
75%                   6.350400e+04    -1.997962e+01    -4.376771e+01
max                   9.999000e+04     4.506593e+01     1.211054e+02

In [24]:
# Non-numeric description
Geolocation.describe(include=['O'])

geolocation_city geolocation_state
count           1000163           1000163
unique             8011                27
top           sao paulo                SP
freq             135800            404268

- This table can be useful to locate the cities in a Dashboard

## Marketing

In [25]:
# Dataset load and samples
Marketing = pd.read_csv('../Datasets/Datasets_original/olist_marketing_qualified_leads_dataset.csv')
Marketing.sample(10)

mql_id first_contact_date  \
4179  80a373ef28ca3b24c02b24b5b9ad6731         2018-03-08   
1220  9153d9c5735bdd40f73b2e94265ab899         2018-04-12   
4619  f30e13dda67413e38d9dec087e89d3ed         2018-01-30   
3168  acd4dbc10fbfd65a25eda61b46fc7820         2018-04-10   
7526  9703918582c179531c480ef3c37e91d3         2018-04-28   
7479  1458e84a7f1a80984de07508b68237d8         2017-08-22   
2563  74b846eb2dcd22fe703bebcd75a3aff1         2017-10-19   
2970  7f22f2ce01a1d2ec3cb03a27a0db321f         2017-11-19   
5357  972d8bb6ba69dafce218d8c9218de7dd         2018-02-25   
1589  c31de872e4b768853e4180258bb2ab00         2018-02-01   

                       landing_page_id          origin  
4179  4b0974ddecd83fd34439a7312a609824     paid_search  
1220  58326e62183c14b0c03085c33b9fdc44          social  
4619  b76ef37428e6799c421989521c0e5077         unknown  
3168  22c29808c4f815213303f8933030604c  organic_search  
7526  58326e62183c14b0c03085c33b9fdc44          social  
7479  3cd2a8301d5395f21cdd3094c9eaa2f4     paid_search  
2563  22c29808c4f815213303f8933030604c  organic_search  
2970  f017be4dbf86243af5c1ebed0cff36a2  direct_traffic  
5357  966098efaef7a4d665be6b0cb617541b  organic_search  
1589  f017be4dbf86243af5c1ebed0cff36a2  organic_search

In [26]:
Marketing.shape

(8000, 4)

In [27]:
Marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   mql_id              8000 non-null   object
 1   first_contact_date  8000 non-null   object
 2   landing_page_id     8000 non-null   object
 3   origin              7940 non-null   object
dtypes: object(4)
memory usage: 250.1+ KB


In [28]:
# Description
Marketing.describe()

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [29]:
# Non-numeric description
Marketing.describe(include=['O'])

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [30]:
# % of null data
Marketing.isna().sum()*100/Marketing.shape[0]

mql_id                0.00
first_contact_date    0.00
landing_page_id       0.00
origin                0.75
dtype: float64

- No significant data for proposed analysis

## Order_items

In [31]:
# Dataset load and samples
Order_items = pd.read_csv('../Datasets/Datasets_original/olist_order_items_dataset.csv')
Order_items.sample(10)

order_id  order_item_id  \
46964   6abfbfca6fe988d700c523b016c31838              1   
13574   1ed9a7f3d298570629e4d027458f4d49              1   
48037   6d23c62a64505799abe7e3195e81aa73              1   
63245   905cf34043925642f36abb99eea3f3d6              1   
52892   785e5848b43e60ee92f14e0e20dca105              1   
5543    0c82a229fc22597c2361656cba8d3230              4   
2029    0494f6deee34c3bcc09e66f364b0e5a4              1   
106975  f318921851480832ec75983410d612d8              1   
83815   be5d9df3cb44dbb469a80581ede808a2              1   
76542   ae33ac637d8999d976657ff3ab5c106a              1   

                              product_id                         seller_id  \
46964   368c6c730842d78016ad823897a372db  1f50f920176fa81dab994f9023523100   
13574   5cf189f48af24c605a16bcdb701df823  4a3ca9315b744ce9f8e9374361493884   
48037   c8cfa622661e047ea2499f8bce4731a5  4b9750c8ad28220fe6702d4ecb7c898f   
63245   ecbe401f10ecd3b822d940d63cc3311e  cd68562d3f44870c08922d380acae552   
52892   5a09313e3a42d69ab0b9fe5cb5e45341  f262cbc1c910c83959f849465454ddd3   
5543    281905b21f01db1009fd7c01b10dd1d4  eed78ac17f7f795a19a709745f00cd4e   
2029    57e089e3103f5cda6a4ce23b77399bdb  aafe36600ce604f205b86b5084d3d767   
106975  952e73f7e8bee61569a32b35c19bc981  7ad32824caee82087b3e2e5f33b1bf32   
83815   3410cbd7df2130ec3cb9300ef0fe3df1  165fc07beebdcb6190fba8a06db2a449   
76542   9eb0c62c8c4159426b700eb10045dc44  a066df7361163e7ddc5d7452dd17448b   

        shipping_limit_date   price  freight_value  
46964   2017-12-13 15:21:45   59.00          17.66  
13574   2017-09-28 02:25:54   65.90          16.22  
48037   2017-12-11 09:12:02  134.90          16.70  
63245   2018-01-31 19:13:19   79.00          16.31  
52892   2018-03-30 16:30:23   14.99           7.39  
5543    2018-05-21 16:31:58  199.50           5.86  
2029    2018-05-23 00:15:20   47.90          16.32  
106975  2018-08-22 17:10:39   59.00          19.60  
83815   2017-09-18 04:50:36   59.00          17.98  
76542   2018-02-20 11:50:27   84.90          15.34

In [32]:
Order_items.shape

(112650, 7)

In [33]:
Order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [34]:
# Description
Order_items.describe()

order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

In [35]:
# Interquantile Range
IQR = Order_items.price.quantile(0.75) - Order_items.price.quantile(0.25)
lower_lim = Order_items.price.quantile(0.25) - 2.5 * IQR
upper_lim = Order_items.price.quantile(0.75) + 2.5 * IQR
outliers_low = Order_items.price < lower_lim
outliers_up = Order_items.price > upper_lim
print("Outliers total:",(len(Order_items.price[outliers_low]) + len(Order_items.price[outliers_up])))
Order_items[~(outliers_low | outliers_up)].shape

Outliers total: 5035


(107615, 7)

In [36]:
# Z-Score
z = np.abs(stats.zscore(Order_items.price))
outliers = Order_items.price[(z > 3)].index
print("Outliers total:", len(outliers))
Order_items.drop(outliers).shape

Outliers total: 1966


(110684, 7)

In [37]:
# Non-numeric description
Order_items.describe(include=['O'])

order_id                        product_id  \
count                             112650                            112650   
unique                             98666                             32951   
top     8272b63d03f5f79c56e9e4120aec44ef  aca2eb7d00ea1a7b8ebd4e68314663af   
freq                                  21                               527   

                               seller_id  shipping_limit_date  
count                             112650               112650  
unique                              3095                93318  
top     6560211a19b47992c3666cc44a7e94c0  2017-07-21 18:25:23  
freq                                2033                   21

In [38]:
# Product with more sales
Order_items[['order_id','product_id']].groupby(['product_id']).count().sort_values('order_id', ascending=False)

order_id
product_id                                
aca2eb7d00ea1a7b8ebd4e68314663af       527
99a4788cb24856965c36a24e339b6058       488
422879e10f46682990de24d770e7f83d       484
389d119b48cf3043d311335e499d9c6b       392
368c6c730842d78016ad823897a372db       388
...                                    ...
6ca99f05e9c1952a6597264e87f01378         1
6ca54e191592a843087a042b0665d7ef         1
6ca5428af5edb5066b1966ad58501b60         1
6ca32b4631431cb1fe976880797a0f5c         1
fffe9eeff12fcbd74a2f2b007dde0c58         1

[32951 rows x 1 columns]

In [39]:
# Amount of $ for the more frecuent products
Order_items[['price','product_id']].groupby(['product_id']).sum().sort_values('price', ascending=False)

price
product_id                                
bb50f2e236e5eea0100680137654686c  63885.00
6cdd53843498f92890544667809f1595  54730.20
d6160fb7873f184099d9bc95e30376af  48899.34
d1c427060a0f73f6b889a5c7c61f2ac4  47214.51
99a4788cb24856965c36a24e339b6058  43025.56
...                                    ...
2e8316b31db34314f393806fd7b6e185      2.99
680cc8535be7cc69544238c1d6a83fe8      2.90
8a3254bee785a526d548a81a9bc3c9be      2.55
310dc32058903b6416c71faff132df9e      2.29
46fce52cef5caa7cc225a5531c946c8b      2.20

[32951 rows x 1 columns]

In [40]:
# Seller with more orders
Order_items[['order_id','seller_id']].groupby(['seller_id']).count().sort_values('order_id', ascending=False)

order_id
seller_id                                 
6560211a19b47992c3666cc44a7e94c0      2033
4a3ca9315b744ce9f8e9374361493884      1987
1f50f920176fa81dab994f9023523100      1931
cc419e0650a3c5ba77189a1882b7556a      1775
da8622b14eb17ae2831f4ac5b9dab84a      1551
...                                    ...
5f57db27027655e6c6a391601daa0258         1
5e25c632f9db6cfb8bc100421c2737b6         1
5e106d93b717c2682b9383da480364a1         1
d36b07fd3c5088e477a72d1fccfe29d4         1
7d81e74a4755b552267cd5e081563028         1

[3095 rows x 1 columns]

In [41]:
# Seller with more incomes
Order_items[['price','seller_id']].groupby(['seller_id']).sum().sort_values('price', ascending=False)

price
seller_id                                  
4869f7a5dfa277a7dca6462dcf3b52b2  229472.63
53243585a1d6dc2643021fd1853d8905  222776.05
4a3ca9315b744ce9f8e9374361493884  200472.92
fa1c13f2614d7b5c4749cbc52fecda94  194042.03
7c67e1448b00f6e969d365cea6b010ab  187923.89
...                                     ...
34aefe746cd81b7f3b23253ea28bef39       8.00
702835e4b785b67a084280efca355756       7.60
1fa2d3def6adfa70e58c276bb64fe5bb       6.90
77128dec4bec4878c37ab7d6169d6f26       6.50
cf6f6bc4df3999b9c6440f124fb2f687       3.50

[3095 rows x 1 columns]

## Order_payments

In [42]:
# Dataset load and samples
Order_payments = pd.read_csv('../Datasets/Datasets_original/olist_order_payments_dataset.csv');
Order_payments.sample(10)

order_id  payment_sequential payment_type  \
42296  1c38140d4542f0ac92eeafa89c9575c0                   1       boleto   
25099  10990e23642104aa4b9308fbf816611b                   1       boleto   
71383  4cc5e60f36ca3bdbb07ab8a3c490e15d                   1  credit_card   
70501  d6bb1f2d6bbaa5c7ed01225a4eab7dc2                   1  credit_card   
89152  33083ef1b98c3ffaf1c217ec67963aeb                   1  credit_card   
15146  9203da3a0188ae9cca20e56ccbb1bb93                   1  credit_card   
79821  3c3757e099924a1ef846889be651f146                   1       boleto   
92594  aae50e3ecf0a52e135b59e3df0d18c84                   1  credit_card   
91350  34b1c1b1fca85eecadd97c4612423897                   1  credit_card   
73503  144c41d3111c8d2b0bdbf9232f2e0a78                   1  credit_card   

       payment_installments  payment_value  
42296                     1          36.78  
25099                     1          59.73  
71383                     1          69.31  
70501                     1         114.78  
89152                     1          55.70  
15146                     5          57.94  
79821                     1         317.07  
92594                     2          92.26  
91350                     1          30.23  
73503                     3         127.33

In [43]:
Order_payments.shape

(103886, 5)

In [44]:
Order_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [45]:
# Description
Order_payments.describe()

payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

In [46]:
# Non-numeric description
Order_payments.describe(include=['O'])

order_id payment_type
count                             103886       103886
unique                             99440            5
top     fa65dad1b0e818e3ccc5cb0e39231352  credit_card
freq                                  29        76795

- The utility of this table is going to depend on the PO

## Order_reviews

In [47]:
# Dataset load and samples
Order_reviews = pd.read_csv('../Datasets/Datasets_original/olist_order_reviews_dataset.csv')
Order_reviews.sample(10)

review_id                          order_id  \
88898  f84eb3d3cab99b78af3b48ebabf8eb78  d5d94fa0d1e2f561eb38c08dd582cad8   
95137  36a8a4854a93689d163e4221e6cb7ef7  d52b1795d584c884bb9c831365a62599   
19132  c6fe95393d2b9d1e0143e8c4b7750a87  9f2d05c96527e5387958a97d0ffe6fff   
72349  20afd7280b34224e3ef776e856a10fd9  037ded3ba29ebe9a96acb1177cc52b51   
17120  a7ae31811cf207828084737d77523b28  318850f283b818e0b9302fdc7d811fde   
27417  a5eca99b47e1ce3d2ed3a5deef76e537  40abd692ba634788bde537d1aff36222   
35649  5cb33274cef7409dcf18e56f4fb79c66  00f2c876aa08fbba04199823952e96c1   
25000  e0a3c5ef82be25048f998dba7a7f9c5f  c8d355d9b0027d16b4f22804df2bbc2d   
1724   85292b92c5a94c273a1783b515a54bd1  ea32e58b3d3a0cfce1bf8ead329aca44   
68779  bbd25aae9271d9831816d19f2efb6deb  cfc558750a7ad84d94c1348dfd89543c   

       review_score review_comment_title  \
88898             3                  NaN   
95137             5                  NaN   
19132             5           recomendo!   
72349             5                  NaN   
17120             1                  NaN   
27417             5                  NaN   
35649             3                  NaN   
25000             1                  NaN   
1724              4                  NaN   
68779             4                  NaN   

                                 review_comment_message review_creation_date  \
88898                                               NaN  2018-07-12 00:00:00   
95137  gostei entregue antes previsto produto excelente  2017-12-06 00:00:00   
19132                                            lindos  2018-05-11 00:00:00   
72349                                       Muito bom!   2017-11-04 00:00:00   
17120                Comprei 3 conjuntos, veio apenas 1  2018-05-16 00:00:00   
27417                                               NaN  2018-05-18 00:00:00   
35649                                               NaN  2017-04-19 00:00:00   
25000                                               NaN  2017-03-31 00:00:00   
1724                                                NaN  2017-05-03 00:00:00   
68779                                               NaN  2018-01-12 00:00:00   

      review_answer_timestamp  
88898     2018-07-12 17:36:21  
95137     2017-12-06 18:45:31  
19132     2018-05-14 11:54:41  
72349     2017-11-05 17:49:31  
17120     2018-05-16 18:28:52  
27417     2018-05-20 22:51:20  
35649     2017-04-22 10:21:21  
25000     2017-03-31 10:15:32  
1724      2017-05-04 17:28:12  
68779     2018-01-13 01:21:36

In [48]:
Order_reviews.shape

(99224, 7)

In [49]:
Order_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [50]:
# Description
Order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [51]:
# Non-numeric description
Order_reviews.describe(include=['O'])

review_id                          order_id  \
count                              99224                             99224   
unique                             98410                             98673   
top     7b606b0d57b078384f0b58eac1d41d78  c88b1d1b157a9999ce368f218a407141   
freq                                   3                                 3   

       review_comment_title review_comment_message review_creation_date  \
count                 11568                  40977                99224   
unique                 4527                  36159                  636   
top               Recomendo              Muito bom  2017-12-19 00:00:00   
freq                    423                    230                  463   

       review_answer_timestamp  
count                    99224  
unique                   98248  
top        2017-06-15 23:21:05  
freq                         4

- The utility of this table is going to depend on the KPIs

## Orders

In [52]:
# Dataset load and samples
Orders = pd.read_csv('../Datasets/Datasets_original/olist_orders_dataset.csv')
Orders.sample(10)

order_id                       customer_id  \
70223  e546da8f5a68e30c67e5cdbb75a3447f  607e4ac7c659ddc86466344b63f73413   
13374  f2fd4e0fc194bc29e11d7d3f83151996  69af39aa7ea31fa28a84d12409e4a019   
42045  8341e99159d0cde5b28d7915b011ce2e  a66712c31e8c65998902f67af30ec475   
37547  be32306676f1b65193ce87367ecf3e34  0cbfb11934fdfab3cd95e2224b6c84f3   
33709  e7374ec864786dbea6a5478d93332e1b  ba3e8c7215f0c4648bf0811d2cc50472   
45276  869801fc99307b961a244f5828e568bc  d989bb99a2ac80add9e16a77da2e009c   
36748  0a393be90b03100dcea4aa21f2e027ec  175cd17c9e0cedbb8234469ec003c4a6   
13466  6409da66534414b64138b8202ac53bc7  a80352353b462d86640b2ff29d16a7db   
43880  94f94ebc2e26aeea2aa9e8945cb9fb0c  e00a5ec8d72cb3c5ed58f50fb1a55cbf   
72098  32720c0c8b42f4c2cdb07cee5ec2444b  8df8c71565b1b7c5d8bec24a7052ba2d   

      order_status order_purchase_timestamp    order_approved_at  \
70223    delivered      2017-08-04 16:11:22  2017-08-04 16:25:16   
13374    delivered      2017-12-23 18:17:06  2017-12-23 18:29:08   
42045    delivered      2017-12-14 14:32:32  2017-12-16 02:36:39   
37547    delivered      2017-11-06 15:04:50  2017-11-07 07:30:34   
33709    delivered      2017-08-19 22:11:32  2017-08-19 22:30:06   
45276    delivered      2018-06-17 09:41:10  2018-06-17 09:55:15   
36748    delivered      2017-11-18 16:04:57  2017-11-18 16:26:11   
13466   processing      2017-03-28 16:59:13  2017-03-29 02:05:15   
43880    delivered      2017-04-17 14:31:24  2017-04-17 14:45:30   
72098    delivered      2017-11-24 22:46:55  2017-11-25 01:52:03   

      order_delivered_carrier_date order_delivered_customer_date  \
70223          2017-08-07 20:21:54           2017-08-15 18:09:36   
13374          2017-12-26 17:32:27           2018-01-04 21:41:37   
42045          2017-12-16 18:03:03           2017-12-21 18:03:38   
37547          2017-11-08 20:26:48           2017-11-28 21:58:39   
33709          2017-08-21 19:03:55           2017-08-28 23:49:53   
45276          2018-06-18 14:29:00           2018-06-19 20:02:56   
36748          2017-11-22 00:16:08           2017-12-05 15:25:32   
13466                          NaN                           NaN   
43880          2017-04-19 13:53:09           2017-05-04 15:18:33   
72098          2017-11-28 18:41:04           2017-12-07 21:34:06   

      order_estimated_delivery_date  
70223           2017-08-28 00:00:00  
13374           2018-01-24 00:00:00  
42045           2018-01-15 00:00:00  
37547           2017-11-29 00:00:00  
33709           2017-09-12 00:00:00  
45276           2018-07-03 00:00:00  
36748           2017-12-15 00:00:00  
13466           2017-04-19 00:00:00  
43880           2017-05-10 00:00:00  
72098           2017-12-22 00:00:00

In [53]:
# New column added: time to complete the delivery
Orders['Delivery_time'] = pd.to_datetime(Orders["order_approved_at"]) - pd.to_datetime(Orders['order_delivered_customer_date'])

# Transform data into days
Orders['Delivery_time'] = -(Orders['Delivery_time'].apply(lambda x: x.days + (x.seconds // 86400)))

In [54]:
Orders.shape

(99441, 9)

In [55]:
Orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 9 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   order_id                       99441 non-null  object 
 1   customer_id                    99441 non-null  object 
 2   order_status                   99441 non-null  object 
 3   order_purchase_timestamp       99441 non-null  object 
 4   order_approved_at              99281 non-null  object 
 5   order_delivered_carrier_date   97658 non-null  object 
 6   order_delivered_customer_date  96476 non-null  object 
 7   order_estimated_delivery_date  99441 non-null  object 
 8   Delivery_time                  96462 non-null  float64
dtypes: float64(1), object(8)
memory usage: 6.8+ MB


In [56]:
# Description
Orders.describe()

Delivery_time
count   96462.000000
mean       12.642844
std         9.520822
min        -6.000000
25%         7.000000
50%        10.000000
75%        16.000000
max       209.000000

In [57]:
# Non-numeric description
Orders.describe(include=['O'])

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

- La utilidad de esta tabla dependerá del objetivo.

## Products

In [58]:
# Dataset load and samples
Products = pd.read_csv('../Datasets/Datasets_original/olist_products_dataset.csv')
Products.sample(10)

product_id      product_category_name  \
12043  672e757f331900b9deea127a2a7b79fd  agro_industria_e_comercio   
29926  1e10b292e5af5e5af60b4e8bc7726e22               beleza_saude   
2680   cdbcde669141ea6af7193857333db4ff           moveis_decoracao   
29215  f0d32532e5d3c1acbf87f4bc139d59cc                  telefonia   
19236  b31f144164588b921d0dd301c3a3f78b                 automotivo   
30194  746d236f81b4ae9d259030ace2833590               beleza_saude   
31884  92e16cf2fab490ab9ae87a305d5f8f7f                 perfumaria   
5203   87a9863375a0a462b168cb02bdad360c                      bebes   
10600  d3e38ef1760a26d888e58942e16b0a70         relogios_presentes   
9449   fe40f7377a50aba380fb2fca57cf730c                   pet_shop   

       product_name_lenght  product_description_lenght  product_photos_qty  \
12043                 39.0                       383.0                 1.0   
29926                 56.0                       656.0                 1.0   
2680                  50.0                       279.0                 2.0   
29215                 35.0                       356.0                 4.0   
19236                 48.0                       691.0                 1.0   
30194                 52.0                      1336.0                 1.0   
31884                 20.0                       372.0                 1.0   
5203                  58.0                      1083.0                 1.0   
10600                 53.0                       641.0                 1.0   
9449                  56.0                      2359.0                 6.0   

       product_weight_g  product_length_cm  product_height_cm  \
12043            2650.0               30.0               30.0   
29926             400.0               19.0                7.0   
2680             5550.0               45.0               30.0   
29215             125.0               18.0                3.0   
19236            2800.0               65.0               27.0   
30194           29100.0               50.0               50.0   
31884             425.0               18.0               11.0   
5203              383.0               22.0               14.0   
10600             500.0               18.0               12.0   
9449             1650.0               42.0               22.0   

       product_width_cm  
12043              20.0  
29926              12.0  
2680               45.0  
29215              12.0  
19236              54.0  
30194              11.0  
31884              12.0  
5203               17.0  
10600              17.0  
9449               32.0

In [59]:
Products.shape

(32951, 9)

In [60]:
Products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


In [61]:
# Description
Products.describe()

product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean             48.476949                  771.495285            2.188986   
std              10.245741                  635.115225            1.736766   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32949.000000       32949.000000       32949.000000   
mean        2276.472488          30.815078          16.937661   
std         4282.038731          16.914458          13.637554   
min            0.000000           7.000000           2.000000   
25%          300.000000          18.000000           8.000000   
50%          700.000000          25.000000          13.000000   
75%         1900.000000          38.000000          21.000000   
max        40425.000000         105.000000         105.000000   

       product_width_cm  
count      32949.000000  
mean          23.196728  
std           12.079047  
min            6.000000  
25%           15.000000  
50%           20.000000  
75%           30.000000  
max          118.000000

In [62]:
# Non-numeric description
Products.describe(include=['O'])

product_id product_category_name
count                              32951                 32341
unique                             32951                    73
top     1e9e8ef04dbcff4541ed26657ea517e5       cama_mesa_banho
freq                                   1                  3029

- The utility of this table is going to depend on the KPIs

## Sellers

In [63]:
# Dataset load and samples
Sellers = pd.read_csv('../Datasets/Datasets_original/olist_sellers_dataset.csv')
Sellers.sample(10)

seller_id  seller_zip_code_prefix  \
1057  5b97c06a046398e7005b2b7c0d707a92                   38770   
1204  cfa5ab0e4d3fc212fd0b3410e7081c62                   36500   
1849  557f22c76691849db52d2abccf0015d0                   75650   
3020  7b3f6ce5c6c22e9b96e7581ebf0df6e1                   13710   
1284  058cb5aeb36d7c0fcae20fc85d5e0a59                   21635   
2654  b4ffb71f0cb1b1c3d63fad021ecf93e1                    3880   
858   5343d0649eca2a983820bfe93fc4d17e                    9270   
1018  8936ba0de1073ee888cb3185ae5dc661                    3372   
406   48e954b2da4f6a750c844173b347ed40                   91740   
2535  f783a638e9c6af4a7d0dbfde811e7a81                    3063   

         seller_city seller_state  
1057   joao pinheiro           MG  
1204             uba           MG  
1849       morrinhos           GO  
3020          tambau           SP  
1284  rio de janeiro           RJ  
2654       sao paulo           SP  
858      santo andre           SP  
1018       sao paulo           SP  
406     porto alegre           RS  
2535       sao paulo           SP

In [64]:
Sellers.shape

(3095, 4)

In [65]:
# Most frequent cities
Sellers.seller_city.value_counts()

sao paulo              694
curitiba               127
rio de janeiro          96
belo horizonte          68
ribeirao preto          52
                      ... 
taruma                   1
s jose do rio preto      1
domingos martins         1
messias targino          1
leme                     1
Name: seller_city, Length: 611, dtype: int64

In [66]:
Sellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


In [67]:
# Description
Sellers.describe()

seller_zip_code_prefix
count             3095.000000
mean             32291.059451
std              32713.453830
min               1001.000000
25%               7093.500000
50%              14940.000000
75%              64552.500000
max              99730.000000

In [68]:
# Non-numeric description
Sellers.describe(include=['O'])

seller_id seller_city seller_state
count                               3095        3095         3095
unique                              3095         611           23
top     3442f8959a84dea7ee197c632cb2df15   sao paulo           SP
freq                                   1         694         1849

## Product_category_name_translation

In [69]:
# Dataset load and samples
Product_category_name_translation = pd.read_csv('../Datasets/Datasets_original/product_category_name_translation.csv')
Product_category_name_translation.sample(10)

product_category_name  \
26  moveis_cozinha_area_de_servico_jantar_e_jardim   
56                              eletrodomesticos_2   
25               construcao_ferramentas_construcao   
3                                  cama_mesa_banho   
36              construcao_ferramentas_ferramentas   
1                           informatica_acessorios   
2                                       automotivo   
66                                          flores   
55                          fashion_roupa_feminina   
32                                     eletronicos   

              product_category_name_english  
26  kitchen_dining_laundry_garden_furniture  
56                        home_appliances_2  
25          construction_tools_construction  
3                            bed_bath_table  
36                  costruction_tools_tools  
1                     computers_accessories  
2                                      auto  
66                                  flowers  
55                   fashio_female_clothing  
32                              electronics

In [70]:
Product_category_name_translation.shape

(71, 2)

In [71]:
# Descriptiom
Product_category_name_translation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


In [72]:
# Non-numeric description
Product_category_name_translation.describe(include=['O'])

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

## Valoration table

We create a new table which allow us to give each seller a score, based on different metrics

- Location
- Amount of distinct products
- Type of products
- Time to fill order
- Review score
- Number of orders
- Number of sales
- Income

In [73]:
datasets_combinados=Orders.merge(Order_reviews,on="order_id")
datasets_combinados=datasets_combinados.merge(Order_payments,on="order_id")
datasets_combinados=datasets_combinados.merge(Order_items,on="order_id")
datasets_combinados=datasets_combinados.merge(Sellers,on="seller_id")
datasets_combinados=datasets_combinados.merge(Products,on="product_id")

In [74]:
datasets_combinados.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  128e10d95713541c87cd1a2e48201934  a20e8105f23924cd00833fd87daa0831   
4  0e7e841ddf8f8f2de2bad69267ecfbcf  26c7ac168e1433912a51b924fbd34d34   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3    delivered      2017-08-15 18:29:31  2017-08-15 20:05:16   
4    delivered      2017-08-02 18:24:47  2017-08-02 18:43:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2017-10-04 19:55:00           2017-10-10 21:25:13   
2          2017-10-04 19:55:00           2017-10-10 21:25:13   
3          2017-08-17 15:28:33           2017-08-18 14:44:43   
4          2017-08-04 17:35:43           2017-08-07 18:30:01   

  order_estimated_delivery_date  Delivery_time  \
0           2017-10-18 00:00:00            9.0   
1           2017-10-18 00:00:00            9.0   
2           2017-10-18 00:00:00            9.0   
3           2017-08-28 00:00:00            3.0   
4           2017-08-15 00:00:00            5.0   

                          review_id  ...  seller_city seller_state  \
0  a54f0611adc9ed256b57ede6b6eb5114  ...         maua           SP   
1  a54f0611adc9ed256b57ede6b6eb5114  ...         maua           SP   
2  a54f0611adc9ed256b57ede6b6eb5114  ...         maua           SP   
3  b46f1e34512b0f4c74a72398b03ca788  ...         maua           SP   
4  dc90f19c2806f1abba9e72ad3c350073  ...         maua           SP   

   product_category_name product_name_lenght product_description_lenght  \
0  utilidades_domesticas                40.0                      268.0   
1  utilidades_domesticas                40.0                      268.0   
2  utilidades_domesticas                40.0                      268.0   
3  utilidades_domesticas                40.0                      268.0   
4  utilidades_domesticas                40.0                      268.0   

   product_photos_qty product_weight_g  product_length_cm  product_height_cm  \
0                 4.0            500.0               19.0                8.0   
1                 4.0            500.0               19.0                8.0   
2                 4.0            500.0               19.0                8.0   
3                 4.0            500.0               19.0                8.0   
4                 4.0            500.0               19.0                8.0   

   product_width_cm  
0              13.0  
1              13.0  
2              13.0  
3              13.0  
4              13.0  

[5 rows x 36 columns]

In [75]:
# We modify the selected IDs
columnas = [0,1,9,19,20,21]
datasets_combinados.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date',
       'Delivery_time', 'review_id', 'review_score', 'review_comment_title',
       'review_comment_message', 'review_creation_date',
       'review_answer_timestamp', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value', 'order_item_id', 'product_id',
       'seller_id', 'shipping_limit_date', 'price', 'freight_value',
       'seller_zip_code_prefix', 'seller_city', 'seller_state',
       'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

In [78]:
# We apply LE to reduce storage
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
for col in columnas:
    datasets_combinados[datasets_combinados.columns[col]] = le.fit_transform(datasets_combinados[datasets_combinados.columns[col]])

In [79]:
Valoracion = pd.DataFrame(columns=['seller_id'])

In [81]:
Valoracion.seller_id = datasets_combinados.seller_id.unique()
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'seller_city']].groupby(['seller_id']).max(), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'seller_state']].groupby(['seller_id']).max(), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'product_id']].groupby(['seller_id']).count(), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'product_category_name']].groupby(['seller_id']).nunique(), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'Delivery_time']].groupby(['seller_id']).mean().round(2), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'review_score']].groupby(['seller_id']).mean().round(2), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'order_id']].groupby(['seller_id']).count(), on= 'seller_id')
Valoracion = Valoracion.merge(datasets_combinados[['seller_id', 'price']].groupby(['seller_id']).sum(), on= 'seller_id')

In [82]:
Valoracion.rename({'product_id':'distinct_prod', 'Delivery_time':'delivery_avg', 'product_category_name':'distinct_categories',\
                   'review_score':'review_avg', 'order_id':'total_orders', 'price':'total_income'}, axis=1, inplace=True)

In [83]:
Valoracion.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3090 entries, 0 to 3089
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   seller_id                3090 non-null   int32  
 1   seller_city_x            3090 non-null   object 
 2   seller_state_x           3090 non-null   object 
 3   product_id_x             3090 non-null   int64  
 4   product_category_name_x  3090 non-null   int64  
 5   seller_city_y            3090 non-null   object 
 6   seller_state_y           3090 non-null   object 
 7   product_id_y             3090 non-null   int64  
 8   product_category_name_y  3090 non-null   int64  
 9   Delivery_time            2965 non-null   float64
 10  review_avg               3090 non-null   float64
 11  total_orders             3090 non-null   int64  
 12  total_income             3090 non-null   float64
dtypes: float64(3), int32(1), int64(5), object(4)
memory usage: 325.9+ KB


In [85]:
Valoracion.to_csv('../Datasets/Datasets_resultantes\Valoration.csv')